pkg/gardenlet/controller/shoot/care/add_test.go

```diff
@@ -162,6 +163,12 @@ var _ = Describe("Add", func() {
 				oldShoot.Status.LastOperation.State = gardencorev1beta1.LastOperationStateProcessing
 				Expect(p.Update(event.UpdateEvent{ObjectOld: oldShoot, ObjectNew: shoot})).To(BeTrue())
 			})
+
+			It("should return true when seed gets assigned to shoot", func() {
+				oldShoot := shoot.DeepCopy()
+				shoot.Spec.SeedName = pointer.String("test-seed")
+				Expect(p.Update(event.UpdateEvent{ObjectOld: oldShoot, ObjectNew: shoot})).To(BeTrue())
+			})
```

Comment:
Nit: You could add another test for ensuring `false` is returned when the seed name is changed :)